src/Illuminate/Database/Query/Grammars/MySqlGrammar.php

```diff
@@ -166,18 +166,14 @@ protected function compileJsonUpdateColumn($key, JsonExpression $value)
      */
     public function prepareBindingsForUpdate(array $bindings, array $values)
     {
-        $index = 0;
```

Comment:
Bindings are coming in raw format now, so the processing should be changed accordingly and can be simplified.